savant/deepstream/buffer_processor.py

```diff
@@ -654,9 +661,13 @@ def prepare_element_output(self, element: PipelineElement, buffer: Gst.Buffer):
                                     obj_meta=nvds_obj_meta,
                                     selection_type=ObjectSelectionType.REGULAR_BBOX,
                                 )
-                                nvds_set_obj_uid(
-                                    frame_meta=nvds_frame_meta, obj_meta=nvds_obj_meta
-                                )
+                                try:
+                                    nvds_set_obj_uid(
+                                        frame_meta=nvds_frame_meta,
+                                        obj_meta=nvds_obj_meta,
+                                    )
+                                except UIDError:
```

Comment:
UID was set as `VideoFrame.id` at some point, now it's not. Removed this `try ... except`.

Reply:
UID already set when the objects in the input frame already have attributes: `nvds_add_attr_meta_to_obj` sets the UID. This can be reproduced by connecting one module to another. E.g. update samples/conditional_video_processing/docker-compose.x86.yml and run it:

```yaml
  module-1:
    image: ghcr.io/insight-platform/savant-deepstream:latest
    restart: unless-stopped
    volumes:
      - zmq_sockets:/tmp/zmq-sockets
      - ../../models/peoplenet_detector:/models
      - ../../downloads/peoplenet_detector:/downloads
      - ..:/opt/savant/samples
    command: samples/conditional_video_processing/demo.yml
    environment:
      - ZMQ_SRC_ENDPOINT=sub+bind:ipc:///tmp/zmq-sockets/input-video.ipc
      - ZMQ_SINK_ENDPOINT=pub+connect:ipc:///tmp/zmq-sockets/input-module-2-video.ipc
      - FPS_PERIOD=1000
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]

  module-2:
    image: ghcr.io/insight-platform/savant-deepstream:latest
    restart: unless-stopped
    volumes:
      - zmq_sockets:/tmp/zmq-sockets
      - ../../models/peoplenet_detector:/models
      - ../../downloads/peoplenet_detector:/downloads
      - ..:/opt/savant/samples
    command: samples/conditional_video_processing/demo.yml
    environment:
      - ZMQ_SRC_ENDPOINT=sub+bind:ipc:///tmp/zmq-sockets/input-module-2-video.ipc
      - ZMQ_SINK_ENDPOINT=pub+bind:ipc:///tmp/zmq-sockets/output-video.ipc
      - FPS_PERIOD=1000
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]
```